activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -150,6 +158,14 @@ def decryption_options
         def clean_text_scheme
           @clean_text_scheme ||= ActiveRecord::Encryption::Scheme.new(downcase: downcase?, encryptor: ActiveRecord::Encryption::NullEncryptor.new)
         end
+
+        def reserialize(value)
+          if cast_type.binary?
```

Comment:
Is this the lone case needing reserialization?

Reply:
Text and binary columns are the only ones able to store the encrypted JSON blobs.

I checked whether we could always serialize here instead, in the hope that it would be a no-op for String columns, but type serialization is much more complicated than that - it might be converting serialized or store values for instance.

I've changed this in https://github.com/rails/rails/pull/50920/commits/38b7df7aa2629e27c1afa7a0d2eb2c961077a1aa, now we just wrap strings as Binary::Data, as that's actually reversing what the encryptor does. I've also renamed the method as "serialize" is overloaded with meanings here.